app/controllers/products_controller.rb

```diff
@@ -0,0 +1,46 @@
+class ProductsController < ApplicationController
+  before_action :require_login, only: [:new, :create, :edit, :update, :destroy]
+
+  def self.model
```

Comment:
We believe it's required for the shared controller specs to work properly. Is this correct? Is this a place where a comment would be recommended?